clcache/__main__.py

```diff
@@ -1527,46 +1530,78 @@ def createManifestEntry(manifestHash, includePaths):
 
 
 def main():
-    if len(sys.argv) == 2 and sys.argv[1] == "--help":
-        print("""
-clcache.py v{}
-  --help    : show this help
-  -s        : print cache statistics
-  -c        : clean cache
-  -C        : clear cache
-  -z        : reset cache statistics
-  -M <size> : set maximum cache size (in bytes)
-""".strip().format(VERSION))
-        return 0
+    # These Argparse Actions are necessary because the first commandline
+    # argument, the compiler executable path, is optional, and the argparse
+    # class does not support conditional selection of positional arguments.
+    # Therefore, these classes check the candidate path, and if it is not an
+    # executable, stores it in the namespace as a special variable, and
+    # the compiler argument Action then prepends it to its list of arguments
+    class CommandCheckAction(argparse.Action):
+        def __call__(self, parser, namespace, values, optional_string=None):
+            if values and not values.endswith(".exe"):
```

Comment:
Should the test for `.exe` maybe be case-insensitive (e.g. by making `values` all lower-case first) given that file names on Windows are typically case-insensitive?

Reply:
A point

Reply:
Done